### PR TITLE
AKS: Review request for Microsoft.ContainerService to add version stable/2021-05-01

### PR DESCRIPTION
**Pull request by levimm**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dev-containerservice-Microsoft.ContainerService-2021-05-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by levimm** [comment]
confirmed sdk breaking change is a bug at go-sdk side. This bug will be fixed today.

**Review by gossion — APPROVED** [review]

**Review by feiskyer — APPROVED** [review]

**Comment by levimm** [comment]
Hi ARM reviewer, the CI-BreakingChange-Go is not caused by this pr. Please help review. The individual commits were approved by ARM before.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    "subscriptionId": "subid1",
+    "resourceGroupName": "rg1",
+    "resourceName": "clustername1"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "category": "azure-resource-management",
```
Comment (majastrz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json:13]: >category [](http://example.com/codeflow?start=13&length=8)

Each item in the list response should be modeled like any other ARM resource. This means that it should have an `id`, `name`, and `type` properties. Any RP-specific properties should be included in the `properties` property. (Yes, this is required even though this is a proxy resource.)

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/outboundNetworkDependenciesEndpoints": {

You should also add a single GET path for `outboundNetworkDependenciesEndpoints`.

---
Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json:1876 in 808938d. [](commit_id = 808938d19ab8904ff2aa75d09f29f0b334fecca2, deletion_comment = False)

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json (hunk 1 above)
Comment (gossion) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json:13]: Hi @majastrz ,
This is already answered and signed off in https://github.com/Azure/azure-rest-api-specs/pull/14332#discussion_r629606824 when merging the PR to dev branch

**Comment by gossion** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/outboundNetworkDependenciesEndpoints": {
> > ```
> 
> You should also add a single GET path for `outboundNetworkDependenciesEndpoints`.
> 
> Refers to: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json:1876 in [808938d](https://github.com/Azure/azure-rest-api-specs/commit/808938d19ab8904ff2aa75d09f29f0b334fecca2). [](commit_id = [808938d](https://github.com/Azure/azure-rest-api-specs/commit/808938d19ab8904ff2aa75d09f29f0b334fecca2), deletion_comment = False)

Hi @majastrz ,
This is also answered and signed off in https://github.com/Azure/azure-rest-api-specs/pull/14332#discussion_r629606824 when merging the PR to dev branch

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json (hunk 1 above)
Comment (majastrz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json:13]: Given that you're ok with the suboptimal API design, the restrictions it causes (as explained by @pilor in the linked comment), and the fact that it was already approved by API review, I'll let it slide.

Comment (majastrz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/examples/OutboundNetworkDependenciesEndpointsList.json:13]: Also keep in mind that this is a one-off exception and it should not be used to justify similar designs elsewhere in the RP.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Pull request merged**